### PR TITLE
Remove `SuppressFinalize` from `ClientRemoteSessionDSHandlerImpl`

### PR DESCRIPTION
**Pull request by xtqqczze**
Follow-up to https://github.com/PowerShell/PowerShell/pull/24705.

`GC.SuppressFinalize` should not be invoked for types without a finalizer. `ClientRemoteSessionDSHandlerImpl` is sealed therefore cannot have a finalizer.

### CODE REVIEW

File: src/System.Management.Automation/engine/remoting/client/remotingprotocolimplementation.cs [hunk 1]
```diff
@@ -735,26 +735,12 @@ internal void ProcessNonSessionMessages(RemoteDataObject<PSObject> rcvdData)
...
 
-        /// <summary>
-        /// Public method for dispose.
-        /// </summary>
-        public void Dispose()
-        {
-            Dispose(true);
-
-            GC.SuppressFinalize(this);
-        }
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/remoting/client/remotingprotocolimplementation.cs:746]: The change is not related the PR purpose.

Comment (xtqqczze) [reply src/System.Management.Automation/engine/remoting/client/remotingprotocolimplementation.cs:746]: The implementation is redundant in a sealed class.

Comment (xtqqczze) [reply src/System.Management.Automation/engine/remoting/client/remotingprotocolimplementation.cs:746]: Specifically in a sealed class, `GC.SuppressFinalize` can be removed if the class has no unmanaged resources to dispose.

Comment (xtqqczze) [reply src/System.Management.Automation/engine/remoting/client/remotingprotocolimplementation.cs:746]: https://rules.sonarsource.com/csharp/RSPEC-3234/

Comment (iSazonov) [reply src/System.Management.Automation/engine/remoting/client/remotingprotocolimplementation.cs:746]: Even if it's the right thing to do there is a risk. We do not have exhaustive tests for the remote code. 
I can merge `seal` only.

Comment (xtqqczze) [reply src/System.Management.Automation/engine/remoting/client/remotingprotocolimplementation.cs:746]: Separated the change into https://github.com/PowerShell/PowerShell/pull/24705 so we can merge.

Comment (xtqqczze) [reply src/System.Management.Automation/engine/remoting/client/remotingprotocolimplementation.cs:746]: > Even if it's the right thing to do there is a risk.

Similar changes were merged in https://github.com/PowerShell/PowerShell/pull/15802.

**Review by iSazonov — APPROVED** [review]
